base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -848,17 +849,43 @@ private Result unifyTyMaybeInsert(@NotNull Term upper, @NotNull Result result, E
     return checker.apply(term);
   }
 
-  /// region Helper
+  /**
+   * @param letBinds not empty
```

Comment:
But you actually handled the empty case, right?
I suggest remove this comment.

Reply:
My bad